src/amber.cc

```diff
@@ -28,6 +28,41 @@
 #include "src/vkscript/parser.h"
 
 namespace amber {
+namespace {
+
+const FormatType kDefaultFramebufferFormat = FormatType::kB8G8R8A8_UNORM;
+
+Result GetFrameBuffer(Buffer* buffer, std::vector<Value>* values) {
+  values->clear();
+
+  // TODO(jaebaek): Support other formats
+  if (buffer->AsFormatBuffer()->GetFormat().GetFormatType() !=
+      kDefaultFramebufferFormat) {
+    return Result("GetFrameBuffer Unsupported buffer format");
+  }
+
+  const uint8_t* cpu_memory = static_cast<const uint8_t*>(buffer->GetMemPtr());
+  if (!cpu_memory)
+    return Result("GetFrameBuffer missing memory pointer");
+
+  const auto texel_stride = buffer->AsFormatBuffer()->GetTexelStride();
+  const auto row_stride = buffer->AsFormatBuffer()->GetRowStride();
+
+  for (uint32_t y = 0; y < buffer->GetHeight(); ++y) {
+    for (uint32_t x = 0; x < buffer->GetWidth(); ++x) {
+      Value pixel;
+
+      const uint8_t* ptr_8 = cpu_memory + (row_stride * y) + (texel_stride * x);
+      const uint32_t* ptr_32 = reinterpret_cast<const uint32_t*>(ptr_8);
```

Comment:
I was tempted to ask for an assertion that the byte offset was a multiple of 4.... but we don't even know that the base pointer is at an address aligned to 4bytes. So we're just going to rely on the caller being good.